smoketest/alertlog_test.go

```diff
@@ -0,0 +1,252 @@
+package smoketest
+
+import (
+	"encoding/json"
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/target/goalert/smoketest/harness"
+)
+
+func TestNotificationSentSuccess(t *testing.T) {
+	t.Parallel()
+
+	var sql = makeSQL(false, "SMS", true, true, true)
+	h := harness.NewHarness(t, sql, "add-no-notification-alert-log")
+	defer h.Close()
+
+	// create alert
+	doQL(h, t, makeCreateAlertMut(h), nil)
+	h.Trigger()
+	logs := getLogs(h, t)
+
+	// most recent entry
+	var msg = logs.Alert.RecentEvents.Nodes[0].Message
+	assert.Contains(t, msg, "Notification sent")
+	h.Twilio(t).Device(h.Phone("1")).ExpectSMS("Alert #1: foo")
+}
```

Comment:
On second thought, we can probably make one high-level test, and define our types and methods inside it (so we don't need to worry about namespace pollution)

```go
func TestAlertLog(t *testing.T) {
  t.Parallel()
  const sql = `...`
  type cfg struct {
    CMDisabled bool
    // ...
  }
  type logs struct {
    Alert struct {
      // ...
    }
  }

  check := func(desc string, c cfg, assert func(*testing.T, logs)) {
    t.Run(desc, func(t *testing.T) {
      t.Parallel()
      h := harness.NewHarnessWithData(t, sql, c, "add-no-notification-alert-log")
      defer h.Close()

      doQL(h, t, makeCreateAlertMut(h), nil)
      h.Trigger()
      var l logs
      doQL(h, t, logsQuery, &l)
      assert(t, l)
    })
  }


  check("DisabledContactMethod", cfg{CMDisabled: true, CMType: "SMS", ...}, func(t *testing.T, l logs) {
    assert.Equal(t, "contact method disabled", l.Alert.RecentEvents.Nodes[0].State.Details)
  })
}
```